### PR TITLE
Release botservice microsoft.bot service 2022 09 15

### PR DESCRIPTION
**Pull request by thatChang**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [X] new API version
      - [ ] update existing version for new feature
      - [X] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. Sep/2022
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  Sep/2022
  5. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [X] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[X] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [X] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/botservice/resource-manager/Microsoft.BotService/preview/2021-05-01-preview/botservice.json [hunk 1]
```diff
@@ -1800,11 +1824,13 @@
       "properties": {
         "extensionKey1": {
           "type": "string",
-          "description": "The extensionKey1"
+          "description": "The extensionKey1",
+          "readOnly": true
```
Comment (raosuhas) [inline specification/botservice/resource-manager/Microsoft.BotService/preview/2021-05-01-preview/botservice.json:1828]: Have these breaking changes been signed off in both the older api versions ?

**Comment by raosuhas** [comment]
>           "x-nullable": true

is this a secret ? Should it have x-ms secret ?

---
Refers to: specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/botservice.json:2753 in fea3625. [](commit_id = fea362581836bd57e58a743fce1c2d2b6ec067e1, deletion_comment = False)

**Comment by raosuhas** [comment]
>           "description": "Direct Line embed code of the resource"

update descriptions please

---
Refers to: specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/botservice.json:2772 in fea3625. [](commit_id = fea362581836bd57e58a743fce1c2d2b6ec067e1, deletion_comment = False)

**Comment by raosuhas** [comment]
>           "description": "Direct Line embed code of the resource"

consider using enums instead of booleans as explained here : 

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)
Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

---
Refers to: specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/botservice.json:2772 in fea3625. [](commit_id = fea362581836bd57e58a743fce1c2d2b6ec067e1, deletion_comment = False)

**Comment by raosuhas** [comment]
>           "x-nullable": true

x-ms-secret ?

---
Refers to: specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/botservice.json:2968 in fea3625. [](commit_id = fea362581836bd57e58a743fce1c2d2b6ec067e1, deletion_comment = False)

**Comment by raosuhas** [comment]
>           "x-nullable": true

use enums where applicable instead of string

---
Refers to: specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/botservice.json:2998 in fea3625. [](commit_id = fea362581836bd57e58a743fce1c2d2b6ec067e1, deletion_comment = False)

**Comment by thatChang** [comment]
Fix in progress

File: specification/botservice/resource-manager/Microsoft.BotService/preview/2021-05-01-preview/botservice.json (hunk 1 above)
Comment (thatChang) [reply specification/botservice/resource-manager/Microsoft.BotService/preview/2021-05-01-preview/botservice.json:1828]: Yes

File: specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/examples/UpdateConnection.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          ]
+        }
+      }
+    },
+    "201": {
+      "body": {
+        "id": "/subscriptions/subscription-id/providers/Microsoft.BotService/botServices",
+        "location": "global",
+        "etag": "etag1",
+        "id": "/subscriptions/subscription-id/resourceGroups/OneResourceGroupName/providers/Microsoft.BotService/botServices/samplebotname/connections/sampleConnection",
```
Comment (jianyexi) [inline specification/botservice/resource-manager/Microsoft.BotService/stable/2022-09-15/examples/UpdateConnection.json:60]: this id is duplicated with line 57

**Comment by mentat9** [comment]
@thatChang - Couple of things:
1. This PR has been off of ARM review's radar because you didn't remove the ARMChangesRequested label (comment here; https://github.com/Azure/azure-rest-api-specs/pull/20803#issuecomment-1273856308). Please remove the label in future to facilitate ARM review.
2. New API version PRs must have a base commit that includes all files from the current API version so reviewers can see what's getting changed. Can you submit a new PR that starts with current swagger files (copied unchanged to the new API version folder) and adds changes for the new API version in subsequent commits?
3. It's not recommended to update existing API versions with new stuff from the new API version. It's recommended that customers needing the new stuff move to the new API version. Can you explain why you want to add new properties and other changes back to 3 older API versions?

**Comment by mikekistler** [comment]
I've re-reviewed the breaking changes report and while there appear to be a few new issues reported these all appear to be benign.

**Comment by mentat9** [comment]
@thatChang - I didn't see any response to my comments: https://github.com/Azure/azure-rest-api-specs/pull/20803#issuecomment-1370277643. Note, at minimum before your new API can be reviewed by ARM you will need to provide a PR that has the correct base commit for the new API being added.

**Comment by thatChang** [comment]
@mentat9 I have a new PR https://github.com/Azure/azure-rest-api-specs/pull/22077 with the base required.
For the reason why the old versions were edited. It's because S360 reports issues from multiple versions, including new and old ones. To resolve the issues, we have to do so. The S360 issues were detected from live site requests, we cannot control what version to be used by customer but update anything involved.

**Comment by mentat9** [comment]
> @mentat9 I have a new PR #22077 with the base required. For the reason why the old versions were edited. It's because S360 reports issues from multiple versions, including new and old ones. To resolve the issues, we have to do so. The S360 issues were detected from live site requests, we cannot control what version to be used by customer but update anything involved. However, we are continuing working on reduce usage of old version.
Thank-you @thatChang - I will follow up on that PR.

**Comment by mentat9** [comment]
@thatChang - Please close this PR if it's no longer needed.

**Pull request closed without merging**